crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -181,8 +181,22 @@ impl IRGenerator {
     ) -> NodeId {
         let element_type = self.get_object_type_from_abi(el_type);
         let (v_id, array_idx) = self.new_array(name, element_type, len as u32, ident_def);
-        self.context.mem[array_idx].values = vecmap(witness, |w| w.into());
-        self.context.get_current_block_mut().update_variable(v_id, v_id);
+        let mut values = Vec::new();
+        for (i, w) in witness.iter().enumerate() {
+            let mut var = Variable::new(
+                element_type,
+                format!("{name}_{i}"),
+                None,
+                self.context.current_block,
+            );
+            var.witness = Some(*w);
+            values.push(self.context.add_variable(var, None));
```

Comment:
Can use vecmap here

Reply:
Done, although a for loop is clearer (since it does not use extra feature) and more concise (since it is less lines of code). The for expression with a ";" would be the best! Wish we'd had it in noir :)  
n.b. I used map instead of vecmap due to issue with enumerate.

Reply:
`vecmap` should be clearer and shorter with `witness.iter().enumerate()` as the first argument. What issues were you running into with enumerate?

Reply:
There is _no method named `vecmap` found for struct `Enumerate`_

Reply:
Its a function and not a method, so it should be `vecmap(witness.iter().enumerate(), lambda...)`